src/models/login.ts

```diff
@@ -0,0 +1,43 @@
+import open from "open";
+import express from "express";
+import cors from "cors";
+import { setToken } from "../helper/auth";
+
+
+const login = async (
+    port: number = 4444,
+): Promise<void> => {
+    const authUrl = `https://partner.zid.sa/cli/authorized`;
+    console.log("Opening your browser to login...");
+    await open(authUrl);
+
+    const app = express();
+    app.use(cors());
+    const server = app.listen(port, () => {
+        console.log(`Listening for authentication callback on http://localhost:${port}/auth/callback`);
+    });
+
+    app.get('/auth/callback', (req, res) => {
```

Comment:
Great job, I suggest defined constant links auth link, callback, ...etc in a common place.

Reply:
since it's a single responsibility actions/commands
I didn't see the need for defining them in a common place